services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationController.kt

```diff
@@ -89,7 +102,18 @@ class ApplicationController(
     )
     fun create(@RequestBody application: Application): Application {
         val userId = currentUserId()
+        val hankeTunnus = application.hankeTunnus ?: throw HankeNotFoundException("")
+        val hankeId = hankeService.getHankeId(hankeTunnus)
+
+        if (
+            hankeId == null ||
+                !permissionService.hasPermission(hankeId, userId, PermissionCode.EDIT_APPLICATIONS)
+        ) {
+            throw HankeNotFoundException("")
```

Comment:
Would 403 be better if user does not have permission for this operation?

Reply:
Nope, it's better if the user can't differentiate between hanke that don't exist and hanke they don't have permission to.

Reply:
Well, we could return 403 if we checked that the person a VIEW permission for that hanke.

Reply:
Out of scope for this PR.